content/en/blog/general/using-cel-expressions-in-kyverno-policies/index.md

```diff
@@ -0,0 +1,614 @@
+            kind: ReplicaLimit
+          paramRef:
+            name: "replica-limit-test.example.com"
+            parameterNotFoundAction: "Deny"
+          expressions:
+            - expression: "object.spec.replicas <= params.maxReplicas"
+              messageExpression:  "'Deployment spec.replicas must be less than ' + string(params.maxReplicas)"
+EOF
+```
+The `cel.paramKind` and `cel.paramRef` specify the resource used to parameterize this policy. For this example, it is configured by `ReplicaLimit` custom resources.
+
+The ReplicaLimit could be as follows:
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: rules.example.com/v1
+kind: ReplicaLimit
+metadata:
+  name: "replica-limit-test.example.com"
+maxReplicas: 3
+EOF
+```
+Here’s the corresponding custom resource definition:
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: replicalimits.rules.example.com
+spec:
+  group: rules.example.com
+  names:
+    kind: ReplicaLimit
+    plural: replicalimits
+  scope: Namespaced
+  versions:
+    - name: v1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            apiVersion:
+              type: string
+            kind:
+              type: string
+            metadata:
+              type: object
+            maxReplicas:
+              type: integer
+EOF
+```
+Now, let’s try deploying an app with five replicas.
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx
+spec:
+  replicas: 5
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx-server
+        image: nginx
+EOF
+```
+As expected, the deployment creation will be blocked because it violates the rule.
+```
+Error from server: error when creating "STDIN": admission webhook "validate.kyverno.svc-fail" denied the request: 
+
+resource Deployment/default/nginx was blocked due to the following policies 
+
+check-deployment-replicas:
+  deployment-replicas: Deployment spec.replicas must be less than 3
+```
+Let’s try deploying an app with two replicas.
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx-server
+        image: nginx
+EOF
+```
+The deployment is created successfully. Great!
+```
+deployment.apps/nginx created
+```
+
+## CEL Variables in Kyverno Policies
+If an expression grows too complicated, or part of the expression is reusable and computationally expensive to evaluate. We can extract some parts of the expressions into variables. A variable is a named expression that can be referred later as variables in other expressions.
```

Comment:
Incomplete first sentence here. Did you mean to end with something like "then it cannot be created"? Or is it a recommendation like "then it can be decomposed into variables."?

Reply:
Yes, it is a recommendation. I will modify it.